### ISSUE TITLE
window is not defined 

### ISSUE THREAD

**Issue by amyfranz** [magiclabs/magic-admin-js#109]
I am currently attempting to use version 1.10.0 of the @magic-sdk/admin module in a middleware for a Next.js application running version 13.4.0. 

However, when I import the it, I encounter the following error:

error - window is not defined null error - node_modules/atob/browser-atob.js (44:0) @ <unknown>

Not sure if this is a bug or it is being incorrectly used? 

I noticed a similar closed issue from a year ago https://github.com/magiclabs/magic-admin-js/issues/89

**Comment by nnons** [comment]
+1 on this issue. If these were server-side packages I assume it shouldn't rely on the window object to get anything done. This will be problematic for serverless applications as well.

**Comment by ayv8er** [comment]
Where are you trying to instantiate `@magic-sdk/admin ` in your NextJS application?

**Comment by nnons** [comment]
@ayv8er on the server-side of the application where window is not available. I'm building on Remix which is similar to NextJS. More info: https://remix.run/docs/en/main/route/loader

**Comment by ayv8er** [comment]
I'm unable to speak for Remix as we don't have demonstrated use cases there. But "window is not defined" error is thrown in NextJS when the client-side Magic object is instantiated without checking for window being undefined.
```
import { Magic } from 'magic-sdk';

const createMagic = (key) => typeof window != 'undefined' && new Magic(key);

export const magic = createMagic(process.env.NEXT_PUBLIC_MAGIC_PUBLISHABLE_KEY);
```

**Comment by Blackglade** [comment]
+1 on this issue as well. Experiencing this on both the previous and current release of @magic-sdk/admin. This code seems to fail when declared in my middleware.ts file, but works in my api route file.

```js
const magicAdmin = await Magic.init(process.env.MAGIC_SECRET_KEY)
```
So strange... Any idea what's going on?

<img width="844" alt="Screenshot" src="https://github.com/magiclabs/magic-admin-js/assets/6363077/bf6ee0d3-8e19-4904-b933-0318a1241476">

**Comment by cbonaco1** [comment]
@ayv8er appreciate your response; however I am experiencing the same issue as well. your suggestion was related to instantiating `magic-sdk`, which my understanding is a _**client side**_ sdk, where the issues reported are for `@magic-sdk/admin` which is a _**server-side**_ sdk. as others pointed out, it's strange to me that a server-side library would be checking for `window` at all. 

I have a basic route handler setup for login:
```javascript
// src/app/api/login/route.tsx
import { Magic } from '@magic-sdk/admin';
import { NextRequest, NextResponse } from 'next/server';
const mAdmin = new Magic(process.env.MAGIC_SECRET_KEY);

export async function POST(req: NextRequest) {
  try {
    const didToken = mAdmin.utils.parseAuthorizationHeader(
      req.headers.get('Authorization') ?? ''
    );
    await mAdmin.token.validate(didToken);
    const metaData = await mAdmin.users.getMetadataByToken(didToken);

   // store metadata in cookie

    return NextResponse.json({
      authenticated: true,
    });
  } catch (error) {
    // return error
  }
}

export const runtime = 'edge';
```
and when I hit this endpoint, i see in the nextjs console:
```
- error node_modules/atob/browser-atob.js (44:0) @ <unknown>
- error window is not defined
```
I tried instantiating `mAdmin` from inside the POST function but still the same issue

Strangely enough, when I remove the edge runtime, it works on my local machine, but fails to build on Cloudflare (using `@cloudflare/next-on-pages`).

- `next`: 13.4.4
- `@magic-sdk/admin`: 2.0.0
- `magic-sdk`: 19.3.1

I'm hoping you can re-open this issue and take a look.

**Comment by brandonium21** [comment]
+1 ^

**Comment by ayv8er** [comment]
Reopening this issue and will elevate to our eng team to investigate.

**Comment by Ethella** [comment]
Hey fellas. Thx for reporting this. Taking a look now.

**Comment by cbonaco1** [comment]
@Ethella thanks for addressing this! looks like your PR was merged but the build to publish a new version failed
`Error: Input required and not supplied: token`
If you could take a look and re-run the build to publish a new version that would be great.

**Comment by Ethella** [comment]
@cbonaco1 Noted, I'm currently working with the infra team to get that fixed. Thx for your patience!

**Comment by Ethella** [comment]
:rocket: Issue was released in [`v2.0.1`](https://github.com/magiclabs/magic-admin-js/releases/tag/v2.0.1) :rocket:
